File: blood-bank-be/bloodBank/src/main/java/ftn/uns/ac/rs/bloodbank/appointment/dto/ScheduleAppointmentRequest.java
package ftn.uns.ac.rs.bloodbank.appointment.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ScheduleAppointmentRequest implements Serializable {

    @NonNull
    private UUID appointment_id;
    @NonNull
    private UUID customer_id;

}

File: blood-bank-fe/src/app/model/Requests/ScheduleAppointmentRequest.ts
export interface IScheduleAppointmentRequest {
  appointment_id?: string;
  customer_id?: string;
}
export class ScheduleAppointmentRequest implements IScheduleAppointmentRequest {
  appointment_id?: string;
  customer_id?: string;
  constructor(data?: IScheduleAppointmentRequest){
    if (data) {
      for (const property in data) {
        if (data.hasOwnProperty(property))
          (<any>this)[property] = (<any>data)[property];
      }
    }
  }
}
